contracts/modules/assessment/Assessment.sol

```diff
@@ -189,20 +217,22 @@ contract Assessment is IAssessment, MasterAwareV2 {
     address destination,
     uint104 batchSize
   ) internal returns (uint withdrawn, uint withdrawnUntilIndex) {
-    require(
-      IMemberRoles(internalContracts[uint(ID.MR)]).checkRole(
-        destination,
-        uint(IMemberRoles.Role.Member)
-      ),
-      "Destination address is not a member"
+    bool isMember = IMemberRoles(internalContracts[uint(ID.MR)]).checkRole(
+      destination,
+      uint(IMemberRoles.Role.Member)
     );
+    if (!isMember) {
+      revert NotMember(destination);
+    }
```

Comment:
this checks has the advantage of failing early but also adds gas as it duplicates the check that TokenController does on TokenController.mint:

```solidity
  function _mint(address _member, uint _amount) internal {
    require(
      _member == address(this) || token.whiteListed(_member),
      "TokenController: Address is not a member"
    );
    token.mint(_member, _amount);
  }
```

do we still want to keep it?

Reply:
Back when it was added [aff26b8d](https://github.com/NexusMutual/smart-contracts/commit/6ac83abad6bca096537684c73232b966aff26b8d#diff-fe843a8caf1022f4bc4e945156cb3f5042b4b00eed433a0509ffb4feb334fc31R199), the check in `TokenController` [didn't exist yet](https://github.com/NexusMutual/smart-contracts/blob/6ac83abad6bca096537684c73232b966aff26b8d/contracts/modules/token/TokenController.sol#L101). It seems redundant now. I'm not a fan of early reverts anymore because normally reverts are caught at simulation phase (eth_call or eth_estimateGas) hence they don't consume gas.